resources/metadata/mavenBuild.yaml

```diff
@@ -85,7 +88,75 @@ spec:
         default: false
         aliases:
           - name: maven/createBOM
-
+      - name: altDeploymentRepositoryPassowrd
+        type: string
+        description: Password for the alternative deployment repository to which the project artifacts should be deployed ( other than those specified in <distributionManagement> ). This password will be updated in settings.xml . When no settings.xml is provided a new one is created corresponding with <servers> tag
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: custom/repositoryPassword
+          - name: altDeploymentRepositoryPassowrdId
+            type: secret
```

Comment:
You missed to define this secret:

```
specs:
  inputs:
    secrets:
      - name: sonarTokenCredentialsId
        type: jenkins
        description: "Jenkins 'Secret text' credentials ID containing the token used to authenticate
          with the Sonar Server."
```

Reply:
@CCFenner , my bad missed this, the last push should resolve this